server.js

```diff
@@ -24,9 +16,50 @@ app.use(bodyParser.json())
 
 // Start defining your routes here
 app.get('/', (req, res) => {
-  res.send('Hello world')
+  res.send('Hello world, my first backend!')
+})
+
+
+// Show tops songs 
+// Can filter on queries
+app.get('/song', (req, res) => {
+  const genre = req.query.genre
+  const popularity = req.query.popularity
+
+  let filteringMusic = topMusicData
+
+  // Filter on genre
+  if (genre) {
+    filteringMusic = filteringMusic.filter(song => {
+      const genreTitle = song.genre
+      return genreTitle.includes(genre)
+    }
+  )}
+
+  // Filter on popularity, high or low
+  if (popularity) {
+    if (popularity === 'high') {
+      filteringMusic.sort((a, b) => b.popularity - a.popularity)
+    } else if (popularity === 'low') {
+      filteringMusic.sort((a, b) => a.popularity - b.popularity)
+    }
+  } 
+
+  res.json(filteringMusic)
 })
 
+// Shows specific song with info
+app.get('/song/:id', (req, res) => {
+  const id = req.params.id
+
+  let showSongs = topMusicData.filter((song) =>
+    song.id === +id)
+
+  res.json(showSongs)
+})
```

Comment:
Nice! Using `filter()`, you will get a filtered array as a return. If you instead use `find()`, you would get a single value from the array, and you should be able to use that to check if the id (song) exists or not:

```
app.get("/songs/:id", (req, res) => {
  const id = req.params.id
  let showSongs = topMusicData.find((song) => song.id === +id)
  if (showSongs) {
    res.json(showSongs)
  } else {
    res.status(404).send(`No song found with id: ${id}`)
  }
})
```

Reply:
The variable showSongs could be more descriptive of what it contains - it just returns a single song, so why not call it song 😎